### PR TITLE
resent.py for MXNet-master

### PR DESCRIPTION
**Pull request by jinboci**
Resnet for MXNet-master
Passed the resnet tests in test_model_zoo.py on my local sever.
Changes due to mxnet-master's different implementations from 1.7:
- nn.HybridSequential on mxnet-master does not accept the parameter "prefix"
- No "name_scope" defined in class Block on mxnet-master (block.py)
- Version check errors in version.py
- reset_ctx becomes a member function for class Block instead of ParameterDict (mxnet-master does not have the class ParameterDict)

### CODE REVIEW

File: gluoncv/model_zoo/resnet.py [hunk 1]
```diff
@@ -97,28 +98,27 @@ def __init__(self, channels, stride, downsample=False, in_channels=0,
             self.se = None
 
         if downsample:
-            self.downsample = nn.HybridSequential(prefix='')
+            self.downsample = nn.HybridSequential()
```
Comment (zhreshold) [inline gluoncv/model_zoo/resnet.py:101]: reason to remove prefix='' ?

**Review by zhreshold — COMMENTED** [review]
see comment, the rest looks good

File: gluoncv/model_zoo/resnet.py (hunk 1 above)
Comment (jinboci) [reply gluoncv/model_zoo/resnet.py:101]: Thank you for your review! I am not quite sure but I find that HybridSequential is defined in `nn/basic_layers.py`. In 1.7, its `__init__` function accepts parameters `prefix` and `params` but in master it no longer accepts parameters. It produced erros when I wrote it as `nn.HybridSequential(prefix='')`

Here is a comparison across branches:

image

**Comment by zhreshold** [comment]
I am okay with the changes, once the CI is passed I can merge!

**Comment by jinboci** [comment]
@zhreshold  
By the way, the current error message is:
```
Traceback (most recent call last):

  File "/var/lib/jenkins/miniconda3/envs/gluon_cv_py3_test/lib/python3.6/site-packages/nose/case.py", line 197, in runTest

    self.test(*self.arg)

  File "/var/lib/jenkins/workspace/workspace/gluon-cv-py3/tests/unittests/common.py", line 41, in test_wrapper

    orig_test(*args, **kwargs)

  File "/var/lib/jenkins/workspace/workspace/gluon-cv-py3/tests/model_zoo/test_model_zoo.py", line 155, in test_imagenet_models

    _test_model_list(models, ctx, x)

  File "/var/lib/jenkins/workspace/workspace/gluon-cv-py3/tests/model_zoo/test_model_zoo.py", line 55, in _test_model_list

    net.reset_ctx(ctx)

AttributeError: 'ResNetV1' object has no attribute 'reset_ctx'

```
I think it may be caused by the older version of MXNet during Unit Tests. The class`HybridBlock` in the master branch does have the attribute `reset_ctx` (`ResNetV1` inherits from `HybridBlock`), while in 1.7 or maybe earlier, `reset_ctx` is only defined in class `ParameterDict` but not `HybridBlock`:

https://github.com/apache/incubator-mxnet/blob/ce1e68260eb3cf9219ae4d59df8bdde7361802ec/python/mxnet/gluon/block.py#L1482

https://github.com/apache/incubator-mxnet/blob/7efb5df7fcaaff5b1ced41b7e718e2146ed57814/python/mxnet/gluon/parameter.py#L471

**Comment by mli** [comment]
Job PR-1473-11 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1473/11/index.html 
Code coverage of this PR:  vs. Master:

**Comment by jinboci** [comment]
Tests have passed only for `resnet.py`; all other tests are disabled.
I find that `mxnet.contrib.quantization` does not exist in mxnet-2.0, however, some currently disabled test cases in `test_model_zoo.py` for other models in `model_zoo` are still using funcitons in `mxnet.contrib.quantization`. I might need to fix it later.

**Review by zhreshold — APPROVED** [review]

**Comment by zhreshold** [comment]
@jinboci Looks good to me at this point, we can keep the momentum and fix unrelated issues later

**Pull request merged**